### DISCUSSION TITLE
Creating zip from existing documents in pocketbase

### DISCUSSION THREAD

**Discussion by starredev** [pocketbase/pocketbase#4240] [Q&A]
I'm looking for a solution to create zip from existing files in pocketbase. Anyone able to help me to sort it out?

**Comment by ganigeorgiev** [comment]
The question is too vague and will depend on a lot of things (including where the files you want to zip are stored).

If you are using the JS `pb_hooks` there is no exposed method at the moment to create zips. One possible workaround for this case is to use the `$os` helpers to invoke the shell zip/tar/etc. command.

If you extending PocketBase with Go, then you can either use the helper [`archive` subpackage](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.21.1/tools/archive) or use directly the std [`archive/zip` Go package](https://pkg.go.dev/archive/zip).

If the above doesn't help, please provide more details about your setup and your use case.

**Reply by starredev** [reply]
Basically I want to use the beforeCreate method to catch all the files I’ve added from the uploader in my frontend and wants a zip to be created and added to pocketbase together with files.
